Objects/unicodeobject.c

```diff
@@ -2393,13 +2395,22 @@ unicode_fromformat_write_cstr(_PyUnicodeWriter *writer, const char *str,
             length++;
         }
     }
-    unicode = PyUnicode_DecodeUTF8Stateful(str, length, "replace", NULL);
-    if (unicode == NULL)
-        return -1;
 
-    res = unicode_fromformat_write_str(writer, unicode, width, -1, flags);
-    Py_DECREF(unicode);
-    return res;
+    if (width < 0) {
+        return unicode_decode_utf8_writer(writer, str, length,
+                                          _Py_ERROR_UNKNOWN, "replace", NULL);
```

Comment:
Should not it be `_Py_ERROR_REPLACE`?

Reply:
Oh right, I missed this "optimization" ;-) Fixed.